pkg/services/sqlstore/migrations/migrations.go

```diff
@@ -46,6 +48,11 @@ func (*OSSMigrations) AddMigration(mg *Migrator) {
 	addUserAuthTokenMigrations(mg)
 	addCacheMigration(mg)
 	addShortURLMigrations(mg)
+	// TODO Delete when unified alerting is enabled by default unconditionally (Grafana v9)
+	if err := ualert.CheckUnifiedAlertingEnabledByDefault(mg); err != nil { // this should always go before any other ualert migration
```

Comment:
If the main reason for this is to decide whether to migrate the dashboard alerts to Grafana 8 alerts I would suggest moving this check at the beginning of [this migration](https://github.com/grafana/grafana/blob/24d4c8a9d1354c58f2eb3578e7bd770cf4ddfc1d/pkg/services/sqlstore/migrations/ualert/ualert.go#L226) and probably also [when removing the Grafana 8 alerts data](https://github.com/grafana/grafana/blob/24d4c8a9d1354c58f2eb3578e7bd770cf4ddfc1d/pkg/services/sqlstore/migrations/ualert/ualert.go#L544). In that way the modifications in the `migrator` are not required (because the session is passed as an argument).
I'm a bit concerned about disturbing the migration order (even more) and the changes in the `migrator` therefore I would suggest not to introduce them if they are not absolutely necessary.

Reply:
This is not possible because functions below need the final value when they decide which migration to add to the list. As you know there are two phases of migration: 
- create a list of migrations that need to be run 
- run migrations.
Grafana 8 alerting migrations use the Enabled flag during the first phase. Thus, this function should be executed in that phase, before even migrations are added to the list for the next phase.

Reply:
OK, as an alternative I would suggest: instead of doing so inside the migrator to introduce a `(*SQLStore).OverrideCfg()` and call it before running the migrations [here](https://github.com/grafana/grafana/blob/24053d83d3db614b6c5ffa10075647172c75c526/pkg/services/sqlstore/sqlstore.go#L65)
I believe that would be clearer.